ot/partial.py

```diff
@@ -0,0 +1,1048 @@
+                                        verbose=False):
+    r"""
+    Returns the partial Gromov-Wasserstein transport between (C1,p) and (C2,q)
+
+    The function solves the following optimization problem:
+
+    .. math::
+        GW = \arg\min_{\gamma} \sum_{i,j,k,l} L(C1_{i,k},C2_{j,l})\cdot
+            \gamma_{i,j}\cdot\gamma_{k,l} + reg\cdot\Omega(\gamma)
+
+        s.t.
+             \gamma\geq 0 \\
+             \gamma 1 \leq a\\
+             \gamma^T 1 \leq b\\
+             1^T \gamma^T 1 = m \leq \min\{\|a\|_1, \|b\|_1\}
+
+    where :
+
+    - C1 is the metric cost matrix in the source space
+    - C2 is the metric cost matrix in the target space
+    - p and q are the sample weights
+    - L  : quadratic loss function
+    - :math:`\Omega`  is the entropic regularization term
+        :math:`\Omega=\sum_{i,j} \gamma_{i,j}\log(\gamma_{i,j})`
+    - m is the amount of mass to be transported
+
+    The formulation of the problem has been proposed in [12].
+
+    Parameters
+    ----------
+    C1 : ndarray, shape (ns, ns)
+        Metric cost matrix in the source space
+    C2 : ndarray, shape (nt, nt)
+        Metric costfr matrix in the target space
+    p : ndarray, shape (ns,)
+        Distribution in the source space
+    q : ndarray, shape (nt,)
+        Distribution in the target space
+    reg: float
+        entropic regularization parameter
+    m : float, optional
+        Amount of mass to be transported (default: min (|p|_1, |q|_1))
+    G0 : ndarray, shape (ns, nt), optional
+        Initialisation of the transportation matrix
+    numItermax : int, optional
+        Max number of iterations
+    tol : float, optional
+        Stop threshold on error (>0)
+    log : bool, optional
+        return log if True
+    verbose : bool, optional
+        Print information along iterations
+
+    Examples
+    --------
+    >>> import ot
+    >>> import scipy as sp
+    >>> a = np.array([0.25] * 4)
+    >>> b = np.array([0.25] * 4)
+    >>> x = np.array([1,2,100,200]).reshape((-1,1))
+    >>> y = np.array([3,2,98,199]).reshape((-1,1))
+    >>> C1 = sp.spatial.distance.cdist(x, x)
+    >>> C2 = sp.spatial.distance.cdist(y, y)
+    >>> np.round(entropic_partial_gromov_wasserstein(C1, C2, a, b,50), 2)
+    array([[0.12, 0.13, 0.  , 0.  ],
+           [0.13, 0.12, 0.  , 0.  ],
+           [0.  , 0.  , 0.25, 0.  ],
+           [0.  , 0.  , 0.  , 0.25]])
+    >>> np.round(entropic_partial_gromov_wasserstein(C1, C2, a, b, 50, m=0.25), 2)
+    array([[0.02, 0.03, 0.  , 0.03],
+           [0.03, 0.03, 0.  , 0.03],
+           [0.  , 0.  , 0.03, 0.  ],
+           [0.02, 0.02, 0.  , 0.03]])
+
+    Returns
+    -------
+    :math: `gamma` : (dim_a x dim_b) ndarray
+        Optimal transportation matrix for the given parameters
+    log : dict
+        log dictionary returned only if `log` is `True`
+
+    References
+    ----------
+    .. [12] Peyré, Gabriel, Marco Cuturi, and Justin Solomon,
```

Comment:
Add ref 29 here and discuss it in the doc. This is not the reference where partial entropic gromov  was proposed

Reply:
done